tests/Unit/NumericalAlgorithms/Spectral/SwshTestHelpers.cpp

```diff
@@ -0,0 +1,111 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/NumericalAlgorithms/Spectral/SWSHTestHelpers.hpp"
+
+#include <boost/math/special_functions/binomial.hpp>
+#include <cmath>
+#include <complex>
+#include <unordered_map>
+
+#include "ErrorHandling/Assert.hpp"
+#include "NumericalAlgorithms/Spectral/SWSHTags.hpp"  // IWYU pragma: keep
+
+namespace Spectral {
+namespace SWSH {
+namespace TestHelpers {
+
+double factorial(size_t arg) noexcept {
+  if (arg <= 1) {
+    return 1;
+  }
+  double factorial_result = 1.0;
+  for (size_t product_term = 1; product_term <= arg; ++product_term) {
+    factorial_result *= static_cast<double>(product_term);
+  }
+  return factorial_result;
+}
+
+std::complex<double> spin_weighted_spherical_harmonic(int s, int l, int m,
```

Comment:
I am not qualified to verify the correctness of these equations, though you should make more things `const` :P